distributions/openhab/src/main/resources/bin/update

```diff
@@ -148,21 +148,22 @@ FileGroup=$(ls -ld "$WorkingDir/userdata" | awk '{print $4}')
 echo "Removing openHAB System Files..."
 mkdir -p "$TempDir/runtime"
 mkdir -p "$TempDir/userdata/etc"
-mv "$WorkingDir/runtime"                                "$TempDir/runtime/"
-mv "$WorkingDir/userdata/etc/all.policy"                "$TempDir/userdata/etc/"
-mv "$WorkingDir/userdata/etc/branding.properties"       "$TempDir/userdata/etc/"
-mv "$WorkingDir/userdata/etc/branding-ssh.properties"   "$TempDir/userdata/etc/"
-mv "$WorkingDir/userdata/etc/config.properties"         "$TempDir/userdata/etc/"
-mv "$WorkingDir/userdata/etc/custom.properties"         "$TempDir/userdata/etc/"
-mv "$WorkingDir/userdata/etc/custom.system.properties"  "$TempDir/userdata/etc/"
-mv "$WorkingDir/userdata/etc/distribution.info"         "$TempDir/userdata/etc/"
-mv "$WorkingDir/userdata/etc/jre.properties"            "$TempDir/userdata/etc/"
-mv "$WorkingDir/userdata/etc/org.apache.karaf"*         "$TempDir/userdata/etc/"
-mv "$WorkingDir/userdata/etc/org.ops4j.pax.url.mvn.cfg" "$TempDir/userdata/etc/"
-mv "$WorkingDir/userdata/etc/profile.cfg"               "$TempDir/userdata/etc/"
-mv "$WorkingDir/userdata/etc/startup.properties"        "$TempDir/userdata/etc/"
-mv "$WorkingDir/userdata/etc/system.properties"         "$TempDir/userdata/etc/"
-mv "$WorkingDir/userdata/etc/version.properties"        "$TempDir/userdata/etc/"
+
+## Go through a list of system files that are stored in the update archive.
+echo "New system filelist required, extracting from zip..."
+unzip -qp "$OutputFile" runtime/bin/userdata_sysfiles.list > "$TempDir/filelist.list" || {
+  echo "System Filelist not found in update, exiting..."
+  exit 1
+}
+while IFS= read -r fileName
+do
+  fullPath="$WorkingDir/userdata/etc/$fileName"
+  if [ -f "$fullPath" ]; then
+    mv "$fullPath" "$TempDir/userdata/etc/"
+  fi
+done < "$TempDir/filelist.list"
+
+mv "$WorkingDir/runtime" "$TempDir/runtime/"
 
 ## We need to keep a backup in case the user modified that file
 cp "$WorkingDir/userdata/etc/org.ops4j.pax.logging.cfg" "$WorkingDir/userdata/etc/org.ops4j.pax.logging.cfg.bak"
```

Comment:
Perhaps `org.ops4j.pax.logging.cfg` should also be added to `userdata_sysfiles.list`? 
Why does the upgrade also backup `org.ops4j.pax.logging.cfg`? Isn't that already done in the backup script?

Reply:
Not sure what to do with it at the moment. `org.ops4j.pax.logging.cfg` should be configurable (at the moment) because there's no way of customising the login. The saving a temporary file was necessary when log4j2 was introduced and users would have found a broken logging system by keeping their current configuration.

- If we move it to the system file list, then users will have to edit their logging configuration each time they update.
- If we keep it how it is, they'll need to swap them back if they've already moved over.
- If we remove the line, users upgrading from 2.1.x will have an invalid logging configuration.
- We can add a simple check here to see if the current version is 2.1.0.

Reply:
It could be added to the system file list when the update script renames it to `org.ops4j.pax.logging.cfg.bak` before it starts overwriting the old files.

It looks like `org.ops4j.pax.logging.cfg` is currently not updated during Docker container updates like in this update script. So logging may be broken after an upgrade for Docker container users. For users reporting such issues see [this community thread](https://community.openhab.org/t/new-docker-upgrade-ability/38585/4?u=wborn).

Hopefully one day user customizations can be retained e.g. by splitting the log4j2 configuration file (https://github.com/openhab/openhab-distro/issues/516).